### ISSUE TITLE
set_variable('name', disabler()) does not perform name = disabler()

### ISSUE THREAD

**Issue by megous** [mesonbuild/meson#5764]
Manual states:

```Assigns a value to the given variable name. Calling set_variable('foo', bar) is equivalent to foo = bar.```

It looks like the disabler makes it ignore the whole set_variable() statement.

https://mesonbuild.com/Reference-manual.html#set_variable

This is a problem, since it breaks using dictionaries and foreach to generate build targets that use disabler().

```
list = {
  'lib1': [dep1, dep2],
  'lib2: [dep3],
}

foreach name, deps : list
  lib = static_library(name,
    'src/' + name + '.c',
    dependencies: deps,
  )
  set_variable('lib_' + name, lib)
endforeach
```

When I later use lib_lib1, I'd expect that it contains a disabler() and disables a build of the executable where it is used. But instead lib_lib1 is undefined variable and the build ERRORs out with.

```
tests/meson.build:...: ERROR: Tried to get unknown variable "lib_lib1".
```

**Comment by megous** [comment]
Fix is something along the lines of this patch:

https://megous.com/dl/tmp/0001-Make-set_variable-name-disabler-do-name-disabler.patch

**Comment by megous** [comment]
Fixed in #5725
